enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,348 @@
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+The cluster administrator can check the status of the IngressController as
+usual:
+
+```shell
+oc -n openshift-ingress-operator get ingresscontrollers/default -o yaml
+```
+
+The `Progressing` status condition will indicate whether the cluster
+administrator must take further action.  The status condition may indicate that
+the cluster administrator needs to delete the Service, like so:
+
+```shell
+oc -n openshift-ingress delete services/router-default
+```
+
+If the cluster administrator deletes the Service, the operator then recreates it
+with internal scope.
+
+
+#### As a cluster administrator, I want to make a private (internal) IngressController public (external)
+
+The cluster administrator can change the scope of an IngressController to be
+internal (private) as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+Again, the cluster administrator can check the IngressController's status
+conditions and may need to delete a Service to complete the change in scope.
+
+#### As a cluster administrator, I want to cancel a change to scope before it is completed
+
+Suppose the cluster administrator has changed the scope of an IngressController
+from "External" to "Internal", as follows:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+```
+
+Now the cluster administrator realizes that the scope should not be changed at
+this time, for example due to the disruption that changing the scope would
+cause.  The cluster administrator can cancel the change by changing
+`spec.endpointPublishingStrategy.loadBalancer.scope` back to its original value:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/private --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"External"}}}}'
+```
+
+#### As the provider of a managed service, I want to automate changing an IngressController's scope
+
+A managed service can initiate the operation and tell the operator to complete
+it automatically by annotating the IngressController and changing the scope,
+using an API call that is equivalent to the following commands:
+
+```shell
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"type":"LoadBalancerService","loadBalancer":{"scope":"Internal"}}}}'
+oc -n openshift-ingress-operator annotate ingresscontrollers/default ingress.operator.openshift.io/auto-delete-load-balancer=
```

Comment:
```suggestion
oc -n openshift-ingress-operator annotate ingresscontrollers/default ingress.operator.openshift.io/auto-delete-load-balancer=true
```
Or some value

Reply:
Would you need to add the annotation first, before the change is made?

Reply:
The annotation value doesn't actually matter.

> Would you need to add the annotation first, before the change is made?

No, so automation could either add the annotation when creating the IngressController or add it when updating the scope.  I'll try to clarify this point.